libraries/classes/User.php

```diff
@@ -133,4 +133,59 @@ public static function isLoggedIn()
       return false;
     }
   }
+
+  public function isOnline()
+  {
+    $uid = $session['uid'];
+    if($session['token'] == self::$token)
+    {
+      $db = new Factory;
+      $mysql = $db->getDBO();
+
+      if($mysql->query("SELECT uid FROM user_key WHERE uid=$uid"))
```

Comment:
I think it always checks whether query runs successfully for not. You need to check num rows.